go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/permitio/permit-golang
 
-go 1.17
+go 1.18
```

Comment:
verify it doesn't break clients with go 1.17

Reply:
that was a try of using generics, i'm reverting it to 1.17